File: 1_year/LGDP/L3/PY/sala/4.py
print('\nDescrição:\n1-Número n\n2-Limite x de n\n3-Valor y que n será multiplicado sucesivamente enquanto for <x\n')
while True:
    x=list(map(int,input(': ').split()))
    if len(x)<1:
        break
    if x[0]<=x[1]/5:
        while x[0]<x[1]:
            print(x[0],end=' ')
            x[0]*=x[2]
        print('\n')
print('Até!\n')

File: 1_year/LGDP/L3/C/sala/4.c
#include<stdio.h>
#include<string.h>
#include<stdbool.h>
#include<stdlib.h>
#include"../funcoes.h"
//
int main(){
    char*inp;
    int*v;
    printf("\nDescricao:\n1-Numero n\n2-Limite de n\n3-Valor y que n sera multiplicado sucesivamente enquanto for <x\n\n");
    for(;;){
        inp=(char*)malloc(1000*sizeof(*inp));
        v=(int*)calloc(3,sizeof(int));
        printf("-> ");
        fgets(inp,1000,stdin);
        string_to_int(inp,v);
        if(v[1]==0)
            break;
        if(v[0]<=v[1]/5){
            for(;v[0]<v[1];){
                printf("%i ",v[0]);
                v[0]*=v[2];
            }
            printf("\n");
        }
        printf("\n");
        free(inp);
        free(v);
    } 
    printf("Bye, bye!\n\n");
}